GitUI/UserControls/RevisionGrid/RevisionDataGridView.cs

```diff
@@ -703,6 +610,45 @@ protected override void OnKeyDown(KeyEventArgs e)
             }
         }
 
+        protected override void WndProc(ref Message m)
+        {
+            RepaintFilter(m);
+            base.WndProc(ref m);
+        }
+
+        /// <summary>
+        /// Determines is a forced repaint is needed.
+        /// </summary>
+        /// <remarks>
+        /// In situations where the mouse wheel is spinning fast (for example with free-spinning mouse wheels),
+        /// the message pump is flooded with WM_CTLCOLORSCROLLBAR messages and the DataGridView is not repainted.
+        /// This method injects a WM_PAINT message in such cases to make the GUI feel more responsive.
+        /// </remarks>
+        private void RepaintFilter(Message m)
+        {
+            const int WM_CTLCOLORSCROLLBAR = 0x137;
+            const int WM_PAINT = 0xF;
+
+            if (m.Msg != WM_CTLCOLORSCROLLBAR)
+            {
+                _consequtiveScrollMessageCnt = 0;
+            }
+            else
+            {
+                _consequtiveScrollMessageCnt++;
+            }
+
```

Comment:
Our coding style is to make use of early exit in order to reduce nesting.
```suggestion
                return;
            }

            ++_consequtiveScrollMessageCnt;
```

Reply:
✔ done